go/database/mpt/nodes.go

```diff
@@ -166,6 +167,9 @@ type Node interface {
 	// parameter information and return values see SetValue().
 	ClearStorage(manager NodeManager, thisRef *NodeReference, this shared.WriteHandle[Node], address common.Address, path []Nibble) (newRoot NodeReference, changed bool, err error)
 
+	// HasEmptyStorage returns whether source node has empty storage.
+	HasEmptyStorage(source NodeSource, address common.Address, path []Nibble) (bool, error)
```

Comment:
Instead of adding a method to this interface, the same functionality may be implemented by iterating nodes using this method: https://github.com/Fantom-foundation/Carmen/blob/main/go/database/mpt/nodes.go#L256

I am not 100% sure what is the best option, but it looks to me that we may have over time requests to provide various fields from the nodes and all such getters would do the same - iterate the nodes via descending to a child node, just to reach the requested node and get the field. 

It looks to me as better to use the method mentioned above for this. Wdyt?

Reply:
Good idea. I think this could even simplify the code. Please do check the usage, Im quite not sure whether I used it correctly although the tests pass.